lib/engine.js

```diff
@@ -383,6 +383,15 @@ module.exports = function (s, conf) {
           err.desc = 'could not execute ' + signal + ': error fetching quote'
           return cb(err)
         }
+
```

Comment:
This wont work as buy_pct isn't set at this point...
Furthermore sell doesn't use fees so there is no point refactoring the whole thing.

Simply replace 412:416 with the modified block.

Reply:
> This wont work as buy_pct isn't set at this point...

Right. I'll fix it.

> Simply replace 412:416 with the modified block.

397 line returns 'NaN'.
So trade is not work properly.
Because fee variable is 'undefined'

Reply:
Thats just the linter complaining.

`fee` isn't definite unless a buy order got placed, the block only evaluates on `is_reorder` if s[signal] exists and uses `fee` from the initial run thereafter.

Reply:
> Thats just the linter complaining.

Thats not an just linter complaining
As below, trade is not work properly
```
2018-01-15 23:30:00  14220.93 BTC-USDT   +0.29%      193   67.78 +++       buy  0.00000 BTC  1000.00000000 USDT   +0.00%   -0.08%

2018-01-15 23:29:59 - preparing buy order over 0.06954600 BTC of 989.010 USDT (99%) tradeable balance with a expected fee of 0.98901 USDT (0.1%)

2018-01-15 23:29:59 - placing buy order...

2018-01-15 23:29:59 - buy order placed at 14220.93 USDT

2018-01-15 23:30:03 - 14224.00 vs our 14220.93

2018-01-15 23:30:03 - buy order timed out, adjusting price

2018-01-15 23:30:03 - price changed, resizing order, NaN% remain
2018-01-15 23:45:00  14083.60 BTC-USDT   -0.97%      172   60.91 ++            0.00000 BTC  1000.00000000 USDT   +0.00%   +0.89%
```

Reply:
> NaN% remain

That means reorder_pct isn't set.
What is fee on a reorder?
It either needs to be 0, makerFee or takerFee but cannot be NaN